zebra-network/src/protocol/external/message.rs

```diff
@@ -340,20 +340,29 @@ pub enum Message {
     FilterClear,
 }
 
+/// The maximum size of the rejection message
+const MAX_REJECT_MESSAGE_LENGTH: usize = 12;
+/// The maximum size of the rejection reason
+const MAX_REJECT_REASON_LENGTH: usize = 100;
+
 impl<E> From<E> for Message
 where
     E: Error,
 {
     fn from(e: E) -> Self {
+        let mut message = e.to_string();
+        message.truncate(MAX_REJECT_MESSAGE_LENGTH);
```

Comment:
Can we call `shrink_to_fit` on both strings after we truncate them?
https://doc.rust-lang.org/std/string/struct.String.html#method.shrink_to_fit

If we have a lot of very large error strings, they will take up a lot of memory. That memory isn't freed by `truncate`, we need to call `shrink_to_fit` as well.